ansible_runner/__main__.py

```diff
@@ -652,6 +622,22 @@ def main(sys_args=None):
     )
     add_args_to_parser(isalive_subparser, DEFAULT_CLI_ARGS['positional_args'])
 
+    # streaming commands
+    transmit_subparser = subparser.add_parser(
+        'transmit',
+        help="Send a job to a remote ansible-runner process"
+    )
+    add_args_to_parser(transmit_subparser, DEFAULT_CLI_ARGS['positional_args'])
+
+    worker_subparser = subparser.add_parser(
+        'worker',
+        help="Execute work streamed from a controlling instance"
+    )
+
+    process_subparser = subparser.add_parser(
+        'process',
+        help="Receive the output of remote ansible-runner work and distribute the results"
```

Comment:
I kind of get transmit and worker a little bit, but I don't follow process. Why wouldn't the thing that transmits also process? I see that this takes a logfile, but would it not normally write to the artifacts directory?

Reply:
Originally I implemented a controller and a worker, but it was decided that having the ability to asynchronously process the results would be good.

These subcommands take the `--logfile` flag because that flag is part of the generic args group of flags.  I'm not sure it's meaningful here, as you point out, but I'm also not sure if it would be broken either.